iothub/device/src/Transport/Mqtt/ClientWebSocketChannel.cs

```diff
@@ -94,6 +94,11 @@ protected override void DoDisconnect()
         }
 
         protected override async void DoClose()
+        {
+            await DoCloseAsync().ConfigureAwait(false);
```

Comment:
Same as above for all of these: async void won't be observed - the async actions need to be awaited and the exceptions dispatched to one of the API stacks.